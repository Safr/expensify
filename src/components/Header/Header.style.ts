import styled from 'styled-components';
import { darkBlue, sSize, white, mSize, lSize } from '../../theme/variables';
import media from '../../theme/media';

export const HeaderTag = styled.header`
  width: 100%;
  min-width: 320px;
  margin: 0;
  padding: 0;
  background-color: ${darkBlue};
`;

export const HeaderContainer = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: ${sSize};
  ${media.desktop`
    max-width: 940px;
    margin: 0 auto;
  `}
`;

export const Title = styled.h1`
  margin: 0;
  ${media.tablet`
    margin-bottom: 0;
    margin-right: ${lSize};

    a {
      font-size: ${lSize};
    }
  `}

  a {
    font-size: 24px;
    color: ${white};
  }
`;

export const Button = styled.button`
  outline: none;
  border: none;
  background: none;
  color: ${white};
`;
