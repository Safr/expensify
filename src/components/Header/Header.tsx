import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { Dispatch } from 'redux';
import { startLogout } from '../../actions';
import { Button, HeaderContainer, HeaderTag, Title } from './Header.style';

interface Props {
  onStartLogout: () => Promise<object>;
}

export const Header: React.SFC<Props> = ({ onStartLogout }) => (
  <HeaderTag>
    <HeaderContainer>
      <Title>
        <Link to="/dashboard">Expensify</Link>
      </Title>
      <Button onClick={onStartLogout}>Logout</Button>
    </HeaderContainer>
  </HeaderTag>
);

const mapDispatchToProps = (dispatch: Dispatch<any>) => ({
  onStartLogout: () => dispatch(startLogout()),
});

export default connect(null, mapDispatchToProps)(Header);
