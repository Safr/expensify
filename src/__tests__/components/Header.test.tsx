import React from 'react';
import { shallow } from 'enzyme';
import { Header } from '../../components/Header/Header';
import { Button } from '../../components/Header/Header.style';

describe('Header Component', () => {
  test('should render correctly', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot();
  });
  test('should call startLogout on button click', () => {
    const onStartLogout = jest.fn();
    const wrapper = shallow(<Header onStartLogout={onStartLogout} />);
    wrapper.find(Button).simulate('click');
    expect(onStartLogout).toHaveBeenCalled();
  });
});
