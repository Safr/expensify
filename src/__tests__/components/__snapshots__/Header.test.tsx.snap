// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should render correctly 1`] = `
<styled.header>
  <styled.div>
    <styled.h1>
      <Link
        replace={false}
        to="/dashboard"
      >
        Expensify
      </Link>
    </styled.h1>
    <styled.button>
      Logout
    </styled.button>
  </styled.div>
</styled.header>
`;
